### PR TITLE
[MRG] Nilearn.stats - 3: Reorganize examples from Nistats into Nilearn

### PR DESCRIPTION
**Pull request by kchawla-pi**
To be merged after and pulled from #2304

### CODE REVIEW

**Comment by kchawla-pi** [comment]
@bthirion @GaelVaroquaux 
Nilearn does not document or endorse imports of any submodules. Instead, all the classes and functions are made available as imports at the submodule init itself.

Currently nilearn.stats has submodules. So we would import a function thusly:

```
from nilearn.stats.contrasts import compute_contrasts

compute_contrasts()
```
This approach keeps the functions grouped by modules, and minimize code changes necessary for end-users to migrate existing code from Nistats to nilearn.stats. (I expect there are not too many nistats users for this to be a primary consideration for us though).

The other approach, that Nilearn uses, is to import specific functions at the level of nilearn.stats, to make it
```
from nilearn.stats import compute_contrasts

compute_contrasts()
```
If we are going with this, we will need a list of functions.

Which method are we going to use?

**Comment by kchawla-pi** [comment]
My sphinx-fu is not the best, so anytime anyone wants to jump in to lend a hand, feel free.

**Comment by bthirion** [comment]
I think we should go for the second one: be homogeneous with current nilearn defaults.
Sorry for the additional work !
Thx

**Comment by kchawla-pi** [comment]
> be homogeneous with current nilearn defaults.

Then the functions won't be logically grouped. I think the usability will suffer.

**Comment by kchawla-pi** [comment]
Sphinx Gallery does not support multi-level subdirectories for the example directory. This means we either split the first and second level examples into 2 directories, or create a separate examples gallery for nistats.
I'm going for the first approach.

**Comment by bthirion** [comment]
> Then the functions won't be logically grouped. I think the usability will suffer.

Why ? I find the pattern quite reasonable. But maybe i'm missing something: Do you have an example in mind ?

**Comment by bthirion** [comment]
>  I'm going for the first approach.

Agreed.

**Comment by kchawla-pi** [comment]
@GaelVaroquaux @emdupre @JeromeDockes

Help me figure out the Sphinx problem. I'm out of ideas. I'll be unavailable for a week. If anyone wants to push to this branch to fix this, please do.

**Comment by kchawla-pi** [comment]
Anybody ha any suggestions?

**Comment by emdupre** [comment]
It looks like it's not indexing sub-modules -- maybe because of this line:

```
.. automodule:: nilearn.stats
   :no-members:
```

Could you change it to `:members:` and see if that helps ?

**Comment by kchawla-pi** [comment]
Man these URL errors... can't run CircleCI and test your suggestion.

**Comment by emdupre** [comment]
The latest URL error is a 403, rather than a more temporary error like a 503... Are other PRs also being impacted by this same error ?

It looks like it's specifically for `fetch_haxby`. I'm thinking that the dataset address might need to be updated.

**Comment by kchawla-pi** [comment]
Yeah several other PRs are having the same problem, including the Master branch.

Why must the address be updated? Did the hosts change something without warning?

**Comment by kchawla-pi** [comment]
@emdupre Would you fix the address if that's what's needed?

**Comment by emdupre** [comment]
> @emdupre Would you fix the address if that's what's needed?

I'll look at it now....

**Comment by kchawla-pi** [comment]
Now I am not getting that error, but a different one. Which is progress. Your solution for that error seems to have worked. I don't understand why a generated module is showing up as duplicate. Could it be that `compute_contrast` is now being indexed twice, once as a member function of the module `nilearn.stats` and again from the explicit entry in `references.rst`?

**Comment by kchawla-pi** [comment]
Replacing `:no-members:` with `:members:` gives a different error but also inserts the function signatures of the members, breaking stylistically from the rest of Nilearn. I don't have ideas for this. This is the only problem keeping me from considering this ready for review. @GaelVaroquaux you wanna take a closer look?

**Comment by jeromedockes** [comment]
> Replacing `:no-members:` with `:members:` gives a different error but also inserts the function signatures of the members, breaking stylistically from the rest of Nilearn. I don't have ideas for this. This is the only problem keeping me from considering this ready for review. @GaelVaroquaux you wanna take a closer look?

it seems you also have some genuine test failures?

**Comment by jeromedockes** [comment]
I can reproduce the surf plotting error. not with python3.8 though

**Comment by jeromedockes** [comment]
> I can reproduce the surf plotting error. not with python3.8 though

#2335 should fix it

**Comment by emdupre** [comment]
It looks like like it might be as small as [something like this](https://stackoverflow.com/a/36821191) ? I'm wondering if it might be easier to track this down once #2299 and #2304 are merged and the diff is smaller.

**Comment by bthirion** [comment]
Indeed,; I think we should proceed.
Best,
Bertrand

**Comment by kchawla-pi** [comment]
I would like that, merging #2299 and #2304 now and then revisiting this. Should we do that @GaelVaroquaux

**Comment by bthirion** [comment]
+1

**Comment by kchawla-pi** [comment]
@larsoner If you have ideas, I'm all ears.

**Comment by kchawla-pi** [comment]
Okay, i removed the `:mod:` identifier in the documentation. I think the current solution looks good on the website and is acceptable to merge this PR.
Looking at the website, I am not sure we even need to "fix" this: it doesn't seem broken.

For whatever reasons, now CircleCI is being fussy. I have written to them.
Meanwhile the CircleCI job is here: https://circleci.com/gh/kchawla-pi/nilearn/1058#artifacts/containers/0

And the built documentation direct link is here: https://1058-139860812-gh.circle-artifacts.com/0/doc/_build/html/modules/reference.html#nilearn-stats-first-level-model-first-level-model

I think this is a sufficient milestone to merge at.

2 of Nistats' examples are missing a thumbnail image, but that should be fixed in a subsequent PR (they are missing on the Nistats site as well).
@bthirion @GaelVaroquaux @emdupre Please review this quickly

File: .circleci/config.yml [hunk 1]
```diff
@@ -36,12 +36,12 @@ commands:
             date +%F > today
             date +%U > week_num
       - restore_cache:
-          key: v1-packages+datasets-{{ checksum "week_num" }}-{{ checksum ".circleci/packages-cache-timestamp" }}
-      - restore_cache:
-          key: v1-docs-{{ .Branch }}-{{ checksum "today" }}-{{ checksum ".circleci/docs-cache-timestamp" }}
+          key: v1-packages+datasets-X1GoBe57xEdkvJqbf56Fxw65XP4CSCo5xrUg4kth8tc=
```
[the remark this answers is not in the document]
Comment (kchawla-pi) [reply .circleci/config.yml]: This is temporary so the datasets won't fail due to URLErrors. This is intended to be reverted before merge.

**Comment by GaelVaroquaux** [comment]
Man, 135 files changed. You are a hero!

**Comment by bthirion** [comment]
A big step forward indeed :-)

**Pull request merged**